alicloud/resource_alicloud_cas.go

```diff
@@ -0,0 +1,116 @@
+package alicloud
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cas"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudCas() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceAlicloudCasCreate,
+		Read:     resourceAlicloudCasRead,
+		Update:   resourceAlicloudCasUpdate,
+		Delete:   resourceAlicloudCasDelete,
+		Importer: &schema.ResourceImporter{State: resourceAlicloudCasImporter},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateCasName,
+			},
+			"cert": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCasCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	args := cas.CreateCreateUserCertificateRequest()
+	if v, ok := d.GetOk("name"); ok {
+		args.Name = v.(string)
+	} else {
+		return fmt.Errorf("Need param Name")
+	}
+
+	if v, ok := d.GetOk("cert"); ok {
+		b, err := ioutil.ReadFile(v.(string))
+		if err != nil {
+			return fmt.Errorf("OpenCertificate key file got an error: %#v", err)
+		}
+		args.Cert = string(b)
+	} else {
+		return fmt.Errorf("Need param Cert")
+	}
+
+	if v, ok := d.GetOk("key"); ok {
+		b, err := ioutil.ReadFile(v.(string))
+		if err != nil {
+			return fmt.Errorf("OpenCertificate key file got an error: %#v", err)
+		}
+		args.Key = string(b)
+	} else {
+		return fmt.Errorf("Need param Key")
+	}
+
+	raw, err := client.WithCasClient(func(casClient *cas.Client) (interface{}, error) {
+		return casClient.CreateUserCertificate(args)
+	})
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("CreateUserCertificate got an error: %#v", err)
+	}
```

Comment:
I think the line 74-80 are needless and please removed them.

Reply:
done